tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+  # 3. my_rule_for_numeric_data uses MultiBatchBootstrappedMetricDistributionParameterBuilder to compute multiple parameters to supply ExpectationConfigurationBuilders with pre-computed parameters
+
+  my_rule_for_ids:
+    # Create a value_set based on prior batches (relative to when the profiler is run) and evaluate new batches using this value_set
+    # Rationale: I want alerts when new value_set items are added to my dataset
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from some categorical column domains: vendor_id, rate_code_id, pickup_location_id, dropoff_location_id, payment_type
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _id
+        - _type
+    parameter_builders:
+      - name: my_value_set_for_ids
+        # This parameter is the union of all observed batch value_sets
+        # MultiBatchParameterBuilder does not aggregate Parameters, it returns a list of value_set for each batch
+        class_name: MultiBatchParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_parameter_builder
+        metric_domain_kwargs: $domain.domain_kwargs
+        # Specify all batches prior to the 6 most recent to be used to create this set of Parameters using the below batch_request
+        batch_request:
+          partition_request:
+            partition_index: ":-6"
+        # TODO: Should we also support "offset" from the active_batch (active_batch can be set via passing a validator to the Profiler.profile() method and previously setting the active batch validator.active_batch_id(batch_id="mybatchid"))?
+        # Note: To specify all batches explicitly use the below batch_request
+        # batch_request:
+        #  partition_request:
+        #    partition_index: "0:"
+        # TODO: Maybe this can be omitted and inferred as the default (all batches)? James: maybe default isn't necessarily what's in the validator.
+        metric_name: column.distinct_values
+    expectation_configuration_builders:
+      # Create union of value sets here in the expectation_configuration_builder
+      - expectation_type: expect_column_values_to_be_in_set
+        column: $domain.domain_kwargs.column
+        # Parameter builder returns a list of value_sets so take the union() here
+        # TODO: Build `union(sets: List[Set])` method in GE parsing language
+        # TODO: How to reference list of parameters from ParameterBuilder (proposal below)?
+        value_set: union($parameter.my_value_set_for_ids.parameter_list)
```

Comment:
Could we grab the value_set parameters from a subset of batches that we selected above (e.g. we selected :-6; is it possible to only get a union of value_sets for :-4, or would that need to be its own parameter)? Not that this is necessary here, just trying to get a sense of what operations are possible on a collection of parameters across batches.

Reply:
Hmm great question. I would assume the latter - :-4 would need to be it's own parameter given the current design.
If the parser is able to handle slices, we could do something like `union($parameter.my_value_set_for_ids.parameter_list[:-2])` to get a subset of batches when substituting the parameter (but could not get additional batches as you would have to do going from :-6 to :-4)

Reply:
@anthonyburdi @talagluck Why not instead use a `BatchRequest` with the query that will result in the `Batch` objects that are needed?  In other words, front-load the batch selection to `BatchRequest`, and let the `$parameter.*` syntax be used in a direct way -- we compute parameters and assign the reference to them to `$parameter.*` -- and use all of this value, without additional syntax.  Otherwise, we are getting into the realm of supporting a large syntax.  Do we have use cases to justify such increase in complexity for users to have to learn?

Reply:
Totally agree @alexsherstinsky - that is what I meant by `:-4` currently needing it's own parameter (and it's own `BatchRequest` with filter). The example I gave was purely hypothetical, no use case behind it so we can leave this as a thought experiment and the code as-is for now :)